networkx/algorithms/assortativity/connectivity.py

```diff
@@ -17,7 +17,7 @@ def _avg_deg_conn(G, neighbors, source_degree, target_degree,
     dsum = defaultdict(float)
     dnorm = defaultdict(float)
     for n,k in source_degree(nodes).items():
-        nbrdeg = target_degree(neighbors(n))
+        nbrdeg = target_degree(list(neighbors(n)))
```

Comment:
I think nbrdeg would only be iterated once (in at most one of the three if/elif/elif clauses).  
Am I missing something?